packages/Meshfree/test/PointCloudProblemGenerator/PointCloudProblemGenerator.hpp

```diff
@@ -55,6 +58,8 @@ class PointCloudProblemGenerator
      * \brief Create a general problem where points may exist on multiple
      * processors. Points have a unique global id.
      *
+     * \param field_dim The number of components in the field.
```

Comment:
This input parameter does not make sense to me.  The user can infer the number of components from `src_field.extent( 0 )`.  How many component is up to the derived class.
As for `ExodusProblemGenerator`, I do understand that you need it there because you are not actually reading the fields from the files and you are just allocating memory and filling with zero.  But in that case you can always pass `field_dim` to the constructor.